code/datalab-app/src/app/auth/auth.js

```diff
@@ -113,7 +113,18 @@ function processIdentity(idTokenPayload) {
   return JSON.stringify(pick(idTokenPayload, knownFields));
 }
 
-const AuthZero = new auth0.WebAuth(authConfig);
-const PromisifyAuthZero = Promise.promisifyAll(AuthZero);
-export default new Auth(AuthZero, PromisifyAuthZero);
-export { Auth as PureAuth }; // export for testing
+let authSession;
+
+const initialiseAuth = (authConfig) => {
+  if (!authSession) {
+    const AuthZero = new auth0.WebAuth(authConfig);
+    const PromisifyAuthZero = Promise.promisifyAll(AuthZero);
+
+    authSession = new Auth(AuthZero, PromisifyAuthZero, authConfig);
+  }
+};
+
+const auth = () => (authSession);
```

Comment:
Really sorry, I thought we said that given you had to refactor all usages we were going to call this `getAuth`.
The usage looks kinda strange with `auth()` so might be worth doing?

Reply:
No worries. Yes, you are absolutely right. Will fix.